docs/user-guide/pod-preset/index.md

```diff
@@ -0,0 +1,540 @@
+---
+assignees:
+- jessfraz
+title: Pod Preset
+---
+
+Objects of type `podpreset` allow for injecting certain information into pods
+at creation time such as secrets, volumes, volume mounts and environment
+variables.
+
+See [PodPreset proposal](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/pod-preset.md) for more information.
+
+* TOC
+{:toc}
+
+## What is a Pod Preset?
+
+A _Pod Preset_ is an API resource that allows additional runtime requirements
+for Pods to be injected at creation time based off label selectors.
+
+Using a Pod Preset allows pod template authors to not have to explicitly set
+information for every pod. This way authors of pod templates consuming a
+specific service do not need to know all the details about that service.
+
+## Admission
+
+_Admission control_ with `PodPreset` introspects all incoming pod creation
+requests and injects the pod based off a Selector with the desired attributes.
+
+1. Retrieve all `PodPresets` available for use.
+1. Match the label selector of the `PodPreset` to the pod being created.
+1. Attempt to merge the various defined resources for the `PodPreset` into the
+   Pod being created.
+1. On error, throw an event documenting the merge error on the pod, and create
+   the pod _without_ any injected resources from the `PodPreset`.
+
+### Behavior
+
+This will modify the pod spec. The supported changes to `Env`, `EnvFrom`, and
+`VolumeMounts` apply to the container spec for all containers in the pod with
+the specified matching `Selector`. The changes to Volumes apply to the pod spec
+for all pods matching `Selector`.
+
+The resultant modified pod spec will be annotated to show that it was modified
+by the `PodPreset`. This will be of the form
+`podpreset.admission.kubernetes.io/<pip name>": "<resource version>"`.
+
+
+## Creating a Pod Preset
+
```

Comment:
This is an example; it's not creation instructions. Move the content from "enabling pod preset" up to here; it should go first, because the user needs to know that before they do any of this.